tests/integration/targets/custom_image/tasks/tests.yml

```
@@ -0,0 +1,267 @@
+---
+# This test triggers a bug in the api
+#- name: Delete a non existent custom image by uuid
+#  cloudscale_ch.cloud.custom_image:
+#    uuid: "{{ item }}"
+#    state: absent
+#  loop:
+#    - '415caea5-da7c-4aaa-aaaa-ececd38fb8ea'
+
+- name: Fail delete a non existent custom image by name
+  cloudscale_ch.cloud.custom_image:
+    name: this-image-is-non-existent
+    state: absent
+  register: delete
+- name: Verify Fail delete a non existend custom image by name
+  assert:
+    that:
+      - delete is not changed
+
+- name: Fail import a custom image without url
+  cloudscale_ch.cloud.custom_image:
+    name: "{{ cloudscale_resource_prefix }}-test-image"
+    state: present
+    slug: custom-ansible-image
+    zones: lpg1
+    user_data_handling: 'pass-through'
+    tags:
+      project: mars
+    source_format: raw
+  register: failed_import
+  ignore_errors: true
+- name: Verify Fail import a custom image without url
+  assert:
+    that:
+      - failed_import.msg.startswith('Cannot import a new image without url.')
```

This should also verify that the import indeed fails.
```suggestion
      - failed_import is failed
      - failed_import.msg.startswith('Cannot import a new image without url.')
```